src/Cars.jsx

```diff
@@ -0,0 +1,59 @@
+// src/Cars.jsx
+
+import React, { useContext } from 'react';
+import CarsContext from './context/CarsContext';
+import carBlue from './images/carBlue.jpeg';
+import carRed from './images/carRed.jpeg';
+import carYellow from './images/carYellow.jpeg';
+
+function Cars() {
+  const { redCar, blueCar, yellowCar, setRedCar, setBlueCar, setYellowCar } = useContext(CarsContext);
+    return (
+      <div>
+        <div>
+          <img
+            className={redCar ? 'car-right' : 'car-left'}
+            src={carRed}
+            alt="red car"
+          />
+          <button
+            onClick={() => setRedCar(!redCar)}
+            type="button"
+          >
+            Move
+        </button>
+        </div>
+        <div>
+          <img
+            className={blueCar ? 'car-right' : 'car-left'}
+            src={carBlue}
+            alt="blue car"
+          />
+          <button
+            onClick={() => setBlueCar(!blueCar)}
+            type="button"
+          >
+            Move
+        </button>
+        </div>
+        <div>
+          <img
+            className={yellowCar ? 'car-right' : 'car-left'}
+            src={carYellow}
+            alt="yellow car"
+          />
+          <button
+            onClick={() => setYellowCar(!yellowCar)}
+            type="button"
+          >
+            Move
+        </button>
+        </div>
+      </div>
+    )
+
+};
+
+Cars.contextType = CarsContext;
```

Comment:
Eu ainda estou com muitas dúvidas em hooks! Então, eu queria até confirmar com você, o contextType não seria para usar somente em classe?

Reply:
Essa parte do código já veio pronta no exercício. Eu também tinha entendido que só usa em classe também e acho que é isso mesmo. Para para para usar o contextType o código precisaria de ter um this.context e não tem.